src/provider/model/Model.ts

```diff
@@ -52,6 +52,7 @@ export class Model {
         this._environment.windowClosed.add(this.onWindowClosed, this);
 
         apiHandler.onConnection.add(this.onApiHandlerConnection, this);
+        apiHandler.onDisconnection.add(this.onWindowClosed, this);
```

Comment:
Is this the best way to handle this case? We need to track that the window has lost some of it's context, but we may still want to persist some state.

For example, if a window refreshes or navigates to a new page, shouldn't it stay in the same channel? Removing the window from the model will place it back in the default channel again, which doesn't really feel right to me from a user perspective.

It may take some app integration for that to work nicely, but we should be considering those kind of use-cases (and edge cases) as part of this story.